src/colmap/geometry/gps.h

```diff
@@ -136,10 +137,29 @@ struct PosePrior {
 
   inline bool operator==(const PosePrior& other) const;
   inline bool operator!=(const PosePrior& other) const;
+
+  // Returns the offset applied to the original GPS position.
+  static Eigen::Vector3d Offset();
+  // Returns the "unoffset" position, i.e., the position after removing the
+  // offset.
+  inline Eigen::Vector3d UnoffsetPosition() const;
+
+ private:
+  friend class DatabaseCache;
```

Comment:
Is this really necessary? I don't see any usage of this declaration.

Reply:
I think this is for setting the position_offset. I also got confused. Maybe a setter is better than friend class?

Reply:
OK, I see, it is for the position_offset... I am not so happy with the usage of static here, because it makes this a singleton state. It prevents using this class for two locations in the same process.